File: 2021/day01/solution.py
inp = open("2021/day01/input.txt").read().split("\n")
inp = [int(x) for x in inp]
total = 0
for x in range(1,len(inp)-1):
    if inp[x]>inp[x-1]:
        total += 1

print("P1:", total)
total = 0
for x in range(2,len(inp)-1):
    if (inp[x-1]+inp[x]+inp[x+1])>(inp[x-2]+inp[x-1]+inp[x]):
        total += 1
print("P2:", total)

File: 2021/day01/solution.jl
inp = open(f->read(f, String),"2021/day01/input.txt","r")
letnums = map(f-> parse(Int,f), split(inp,"\n"))
total = 0
for i in range(1,length = length(letnums) -1)
    if letnums[i]<letnums[i+1]
        global total += 1
    end
end
println("P1:",total)
total = 0
for i in range(1, length = length(letnums) -3)
    if letnums[i]<letnums[i+3]
        global total += 1
    end
end
print("P2:",total)